iree/hal/buffer_view.c

```diff
@@ -363,7 +363,6 @@ IREE_API_EXPORT iree_status_t iree_hal_buffer_compute_view_size(
     iree_hal_element_type_t element_type,
     iree_hal_encoding_type_t encoding_type,
     iree_device_size_t* out_allocation_size) {
-  IREE_ASSERT_ARGUMENT(shape);
```

Comment:
make !shape_rank || shape (we still want to check that a pointer is passed if we are rank>0 and going to read from it)